sotodlib/io/load_book.py

```diff
@@ -829,6 +850,10 @@ def _frames_iterator(files, prefix, samples, smurf_proc=None):
         filename = os.path.join(prefix, f)
         offset = i0
 
+        if use_temp_dir:
+            filename, orig_filename = os.path.join(use_temp_dir, 'framefile'), filename
+            shutil.copyfile(orig_filename, filename)
```

Comment:
This can cause issues when two runs require the same observation and have the same tmp dir path. From `copyfile` docs: "If dst already exists, it will be replaced". Potentially this can happen while the file is open.

You can use pathlib to create the destination and check if it exists with something like:

```python
## Assuming use_temp_dir is an absolute path
filename = Path(use_temp_dir, 'framefile')
if not filename.exists():
    shutil.coyfile(...)
```